### DISCUSSION TITLE
how to do a cursorPaginate within the Query Builder

### DISCUSSION THREAD

**Discussion by TheOWler** [rappasoft/laravel-livewire-tables#972] [Q&A]
Hi,
Am using laravel 9 with livewire 2.0
am loking for a solution for big data on the datatable cuz the normal query (with "offset")
Ex: 

> select * from users order by `id` asc limit 10 offset 10;

this query take so long on large data set
so the ultimate solution is cursorPaginate which it implement a query like 

> select * from users where `id` > 10 order by `id` asc limit 10;

so i digged inside vendor\rappasoft\laravel-livewire-tables\src\Traits\WithData.php

and i changed the code like so : 

>use Illuminate\Support\Collection;
.
.
.
 protected function executeQuery()
    {
        if($this->paginationIsEnabled()){
            $result= $this->getBuilder()->cursorPaginate(10) ;
        }
        else{
            $result= $this->getBuilder()->get();
        }
        return $result;
    }

with this implementation am getting this error : **Method Illuminate\Database\Eloquent\Collection::lastPage does not exist.**

can someone help me with this

**Comment by lrljoe** [comment]
Available in V3
